kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/EmbeddedKafkaIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+import uk.gov.homeoffice.digital.sas.config.TestConfig;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumer;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.producer.KafkaProducerService;
+import uk.gov.homeoffice.digital.sas.model.Profile;
+import uk.gov.homeoffice.digital.sas.repository.ProfileRepository;
+
+@SpringBootTest(classes = TestConfig.class)
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+@EmbeddedKafka(
+    partitions = 1,
+    brokerProperties = {
+        "listeners=PLAINTEXT://localhost:3333",
+        "port=3333"
+    }
+)
+class EmbeddedKafkaIntegrationTest {
+
+  private static final String PROFILE_ID = "profileId";
+  private static final String PROFILE_NAME = "Original profile";
+  private static final String UPDATED_PROFILE_NAME = "Updated profile";
+  private static final int CONSUMER_TIMEOUT = 10;
+  private String EXPECTED_CREATE_MESSAGE_PAYLOAD;
+  private Profile profile;
+
+  @Value("${projectVersion}")
+  private String version;
+
+  @Autowired
+  private KafkaProducerService<Profile> kafkaProducerService;
+
+  @Autowired
+  private KafkaConsumer kafkaConsumer;
+
+  @Autowired
+  private ProfileRepository profileRepository;
+
+  @BeforeEach
+  void setup() {
+    profile = new Profile(PROFILE_ID, PROFILE_NAME);
+    EXPECTED_CREATE_MESSAGE_PAYLOAD = generatePayload(version, profile, KafkaAction.CREATE);
+  }
+
+  @Test
+  void shouldSendCreateMessageToTopicFromProducer() throws Exception{
+    // GIVEN
+    kafkaProducerService.sendMessage(profile.getId(), Profile.class, profile, KafkaAction.CREATE);
+
+    // WHEN
+    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
+
+    // THEN
+    assertThat(messageConsumed).isTrue();
+    assertThat(kafkaConsumer.getPayload()).isEqualTo(EXPECTED_CREATE_MESSAGE_PAYLOAD);
+  }
+
+  @Test
+  void shouldSendCreateMessageToTopicWhenProfileIsCreated() throws Exception {
+    // GIVEN
+    profileRepository.save(profile);
+
+    // WHEN
+    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
+
+    // THEN
+    assertThat(messageConsumed).isTrue();
+    assertThat(kafkaConsumer.getPayload()).isEqualTo(EXPECTED_CREATE_MESSAGE_PAYLOAD);
+  }
+
+  @Test
+  void shouldSendUpdateMessageToTopicWhenProfileIsUpdated() throws Exception {
+    // GIVEN
+    profileRepository.save(profile);
+    profile.setName(UPDATED_PROFILE_NAME);
+    profile = profileRepository.save(profile);
+
+    // WHEN
+    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
+
+    // THEN
+    assertThat(messageConsumed).isTrue();
+    String expectedUpdateMessagePayload = generatePayload(version, profile, KafkaAction.UPDATE);
+    assertThat(kafkaConsumer.getPayload()).isEqualTo(expectedUpdateMessagePayload);
+  }
+
+  @Disabled("Currently failing on CI/CD")
```

Comment:
Don't we need to resolve this first?

Reply:
This can be tricky to troubleshoot. So in the interest of keeping things moving, I thought it's better to keep it in and disable it, so it is visible for us to try and resolve in the future. Otherwise, the easy solution would be to remove the test altogether, as it's only affects the delete scenario, which already has unit test coverage

Reply:
might even get resolved in upcomming work for timecard kafka/db transaction testing. As most of the tests in this class would have changed I think.

Reply:
Good point @daleelR

Reply:
I think you've both made good points but I'm struggling to get on board with pushing something out with a test that intermittently fails. How do we not know that this may lead to intermittent failures in the deployed solution.

Reply:
I've borrowed your code to fix the CountDownLatch so that only latest message is returned by the listener